frontend/src/metabase/actions/containers/ActionParametersInputForm/ActionParametersInputForm.tsx

```diff
@@ -0,0 +1,185 @@
+      }).then(setPrefetchValues),
+    [dashboard?.id, dashcard?.id, dashcardParamValues],
+  );
+
+  useEffect(() => {
+    // we need at least 1 parameter value (a PK) to fetch initial values
+    const canPrefetch =
+      Object.keys(dashcardParamValues).length > 0 && dashboard && dashcard;
+
+    if (shouldPrefetch) {
+      setPrefetchValues({});
+      canPrefetch && fetchInitialValues();
+    }
+  }, [
+    shouldPrefetch,
+    dashboard,
+    dashcard,
+    dashcardParamValues,
+    fetchInitialValues,
+  ]);
+
+  const fieldSettings = useMemo(
+    () =>
+      action.visualization_settings?.fields ??
+      // if there are no field settings, we generate them from the parameters and field metadata
+      generateFieldSettingsFromParameters(
+        missingParameters,
+        dashcard?.card?.result_metadata as unknown as Field[],
+      ),
+    [action, missingParameters, dashcard],
+  );
+
+  const initialValues = useMemo(
+    () => getInitialValues(fieldSettings, prefetchValues),
+    [fieldSettings, prefetchValues],
+  );
+
+  const handleSubmit = useCallback(
+    async (params, actions) => {
+      actions.setSubmitting(true);
+      const paramsWithChangedValues = getChangedValues(params, initialValues);
+
+      const { success, error } = await onSubmit(paramsWithChangedValues);
+
+      if (success) {
+        actions.setErrors({});
+        onSubmitSuccess?.();
+
+        shouldPrefetch ? fetchInitialValues() : actions.resetForm();
+      } else {
+        throw new Error(error);
+      }
+    },
+    [
+      onSubmit,
+      onSubmitSuccess,
+      initialValues,
+      fetchInitialValues,
+      shouldPrefetch,
+    ],
+  );
+
+  const hasPrefetchedValues = !!Object.keys(prefetchValues).length;
+
+  if (shouldPrefetch && !hasPrefetchedValues) {
+    return <EmptyState message={t`Choose a record to update`} />;
+  }
+
+  const submitButtonLabel = getSubmitButtonLabel(action);
+
+  const formSettings: ActionFormSettings = action.visualization_settings ?? {
+    type: "button",
+    fields: fieldSettings,
+  };
+
+  return (
+    <ActionForm
+      parameters={missingParameters}
+      formSettings={formSettings}
+      initialValues={initialValues}
+      onClose={onCancel}
+      onSubmit={handleSubmit}
+      submitTitle={submitButtonLabel}
+      submitButtonColor={getSubmitButtonColor(action)}
+    />
+  );
+}
+
+interface ModalProps {
+  onClose: () => void;
+  title: string;
+  showConfirmMessage?: boolean;
+  confirmMessage?: string;
+}
+
+export function ActionParametersInputModal({
+  onClose,
+  title,
+  showConfirmMessage,
+  confirmMessage,
+  ...formProps
+}: ModalProps & Props) {
+  return (
+    <Modal onClose={onClose}>
+      <ModalContent title={title} onClose={onClose}>
+        <>
+          {showConfirmMessage && <ConfirmMessage message={confirmMessage} />}
+          <ActionParametersInputForm {...formProps} />
+        </>
+      </ModalContent>
+    </Modal>
+  );
+}
```

Comment:
Should we keep it in another file maybe? If I needed to patch it, I'd be looking for a file called `ActionParametersInputModal`